backend/prompt_studio/prompt_studio_output_manager/output_manager_helper.py

```diff
@@ -48,11 +53,42 @@ def handle_prompt_output_update(
             # Attempt to update an existing output manager,
             # for the given criteria,
             # or create a new one if it doesn't exist
-            PromptStudioOutputManager.objects.update_or_create(
-                document_manager=document_manager,
-                tool_id=tool,
-                profile_manager=profile_manager,
-                prompt_id=prompt,
-                is_single_pass_extract=is_single_pass_extract,
-                defaults={"output": output, "eval_metrics": eval_metrics},
-            )
+            try:
+                # Create or get the existing record for this document, prompt and
+                # profile combo
+                _, success = PromptStudioOutputManager.objects.get_or_create(
+                    document_manager=document_manager,
+                    tool_id=tool,
+                    profile_manager=profile_manager,
+                    prompt_id=prompt,
+                    is_single_pass_extract=is_single_pass_extract,
+                    defaults={
+                        "output": output,
+                        "eval_metrics": eval_metrics,
+                    },
+                )
+
+                if success:
+                    logger.info(
+                        f"Created record for prompt_id: {prompt.prompt_id} and "
+                        f"profile {profile_manager.profile_id}"
+                    )
+                else:
+                    logger.info(
+                        f"Updated record for prompt_id: {prompt.prompt_id} and "
+                        f"profile {profile_manager.profile_id}"
+                    )
+
+                args: dict[str, str] = dict()
+                args["run_id"] = run_id
+                # Update the record with the run id
+                PromptStudioOutputManager.objects.filter(
+                    document_manager=document_manager,
+                    tool_id=tool,
+                    profile_manager=profile_manager,
+                    prompt_id=prompt,
+                    is_single_pass_extract=is_single_pass_extract,
+                ).update(**args)
```

Comment:
@gaya3-zipstack Possible to avoid making two db calls here?
We could very do a single `get_or_create()` and pass all other args inside `defaults`.

Reply:
No @hari-kuriakose get will use all the args that we pass to it (unique constraints could be a subset here) and come back saying record not found and then create will try to create the record. While creating, if the record, we will see a failure due to integrity constraints.
Eg. get_or_create(key=A, field=v3)
Tables looks like
key     field
A         v1
A         v2
WHile executing get, both filters for key and field wll be applied. We may not find it and so it will go about creating the record. While creating we will see that A already exists and it will throw error

Reply:
@gaya3-zipstack you can use the first value of tuple which is the actual object which is either created or retrived.

and then use obj.run_id = run_id
obj.save()

we can remove the filter query in that case is what i believe.

Reply:
@gaya3-zipstack Anyhow let's proceed for now. We could improvise iteratively.